lib/src/timing.dart

```diff
@@ -139,6 +161,12 @@ class SyncTimeTracker implements TimeTracker {
 
   @override
   Duration get duration => stopTime?.difference(startTime);
+
+  /// Converts to JSON representation
+  ///
+  /// Can't be used before [isFinished]
+  @override
+  Map<String, dynamic> toJson() => _$TimeSliceToJson(this);
```

Comment:
We could extend `TimeSlice` instead of implementing it to pick this up

Reply:
We couldn't. `TimeSlice` has `final` fields, but `SyncTimeTracker` need to be able to change them

Reply:
You're allowed to extend a class with final fields. There are performance implications (end up with two slots on the instance instead of one) but in practices it tends to be OK.

That _might_ change in a future Dart version but no immediate plans.

https://dartpad.dartlang.org/cc751be7684d09170ee0b7719c30d0c2

Reply:
I don't think one line of toJson method worth duplicating the fields

Reply:
sounds good